macaroon.js

```diff
@@ -367,44 +386,65 @@ const Macaroon = class Macaroon {
 };
 
 /**
-  Generates macaroon instances based on the macarono data suppplied.
+  Returns macaroon instances based on the JSON-decoded object
+  in the argument. If this is passed an array, it will decode
+  all the macaroons in the array.
+  TODO accept version 2 format.
   @param {Object|Array} obj A deserialized JSON macaroon or an array of them.
   @return {Macaroon}
 */
-const generateMacaroons = function(obj) {
+const importFromJSONObject = function(obj) {
   if (Array.isArray(obj)) {
-    return obj.map(val => generateMacaroons(val));
+    return obj.map(val => importFromJSONObject(val));
   }
-
-  return new Macaroon({
-    signature: hexToUint8Array(obj.signature),
-    location: obj.location,
-    identifier: obj.identifier,
-    caveats: obj.caveats.map(caveat => {
+  let caveats = []
+  if (obj.caveats !== undefined){
+    caveats = obj.caveats.map(caveat => {
       const _caveat = {
-        _identifier: null,
+        _identifier: requireString(caveat.cid, 'Caveat id.'),
```

Comment:
Do we need the dot at the end of the label?

Reply:
removed